src/main/java/com/openshift/internal/restclient/authorization/AuthorizationContext.java

```diff
@@ -35,12 +46,34 @@ public AuthorizationContext(String token, String expires, IUser user, String sch
 		this.scheme = scheme;
 	}
 	
+	public AuthorizationContext clone() {
+		AuthorizationContext context = new AuthorizationContext(this.token, this.expires, this.user, this.scheme);
+		context.setUserName(this.userName);
+		context.setPassword(this.password);
+		context.setClient(this.client);
+		return context;
+	}
+	
+	public void setClient(IClient client){
+		this.client = client;
+	}
+	
 	@Override
 	public boolean isAuthorized() {
+		if(user == null) {
```

Comment:
we're clearly changing the semantics here which is not a good idea imho. We now possibly do a remote call while before this wasnt the case.
I'd advocate this NOT to do any remote call. I'd let the methods that require a user to do the remote call.and create the user. What is user being used for? I dont see references?